battlesnake-rs/src/mcts_snake.rs

```diff
@@ -65,31 +66,35 @@ impl BattlesnakeFactory for MctsSnakeFactory {
         }
     }
 }
+
 impl<
-        T: Clone
+        GameType: Clone
             + SimulableGame<Instrument, 4>
             + PartialEq
             + RandomReasonableMovesGame
             + VictorDeterminableGame
             + YouDeterminableGame
             + 'static,
-    > MctsSnake<T>
+    > MctsSnake<GameType>
 {
     #[tracing::instrument(
         level = "info",
         skip_all,
         fields(total_number_of_iterations, total_score,)
     )]
-    fn mcts(&self, while_condition: &dyn Fn(&Node<T>, usize) -> bool) -> Result<MoveOutput> {
+    fn mcts<'arena>(
+        &self,
+        while_condition: &dyn Fn(&Node<GameType>, usize) -> bool,
+        arena: &'arena mut Arena<Node<'arena, GameType>>,
+    ) -> &'arena Node<'arena, GameType> {
```

Comment:
We moved the Arena outside these functions so that we can return something that references the arena without running into issues